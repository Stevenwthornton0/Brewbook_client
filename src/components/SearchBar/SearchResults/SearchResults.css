.breweryContainer {
    margin-top: 50px;
}

.SearchResults > h2 {
    text-align: center;
    margin-top: 60px;
    font-size: 28px;
}

@media screen and (max-width: 500px) {


    .SearchResults > h2 {
        font-size: 28px;
    }
}