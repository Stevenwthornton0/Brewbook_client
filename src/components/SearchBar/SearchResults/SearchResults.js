import React, { Component } from 'react';
import BreweryListContext from '../../../contexts/BreweryListContext';
import BreweryList from './BreweryList/BreweryList';
import './SearchResults.css';


class SearchResults extends Component {
    static contextType = BreweryListContext;

    static defaultProps = {
        numResults: 0
    }

    render() {
        const { numResults } = this.props;
        let content
        if (this.context.error) {
            content = <p>There is an error</p>
        } else {
            content = 
                <div className='SearchResults'>
                    <h2>{numResults} Results for {this.props.city}, {this.props.state}</h2>
                    <div className='breweryContainer'>
                        <BreweryList />
                    </div>
                </div>
        }
        return (
            <div>{content}</div>
        )
    }
}

export default SearchResults;

