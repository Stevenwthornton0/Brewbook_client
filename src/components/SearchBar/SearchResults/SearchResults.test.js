import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import SearchResults from './SearchResults';

Enzyme.configure({ adapter: new Adapter() })

describe(`SearchResults smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<SearchResults />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})