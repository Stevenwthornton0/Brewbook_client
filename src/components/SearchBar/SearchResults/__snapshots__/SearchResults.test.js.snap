// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults smoke test renders without crashing 1`] = `
<div>
  <div
    className="SearchResults"
  >
    <h2>
      0
       Results for 
      , 
    </h2>
    <div
      className="breweryContainer"
    >
      <BreweryList />
    </div>
  </div>
</div>
`;
